File: flask/api/v1/admin/analytics/route.py
from flask import Blueprint, jsonify, request
from extensions import db
from sqlalchemy import text
from .top_movies.route import top_movies_bp

analytics_bp = Blueprint('analytics', __name__)

# Register nested blueprint for top movies
analytics_bp.register_blueprint(top_movies_bp, url_prefix='/top-movies')

@analytics_bp.route('/revenue-summary', methods=['GET'])
def get_revenue_summary():
    """Get admin revenue summary from view"""
    # TODO: Add admin role check when authentication is implemented
    # For now, this endpoint is open but should be protected in production
    
    try:
        query = text("SELECT * FROM v_admin_revenue_summary")
        result = db.session.execute(query).first()
        
        if result:
            # Convert row to dictionary
            data = dict(result._mapping)
            
            # Convert Decimal values to float for JSON serialization
            for key, value in data.items():
                if hasattr(value, 'is_integer'):  # Check if it's a Decimal
                    data[key] = float(value)
            
            return jsonify({
                'status': 'success',
                'data': data
            }), 200
            
        return jsonify({
            'status': 'error',
            'message': 'No revenue data available'
        }), 404
        
    except Exception as e:
        return jsonify({
            'status': 'error',
            'message': str(e)
        }), 500

File: movie-ui/app/api/v1/admin/movies/cinemas-screens/route.ts
import { NextResponse } from "next/server"
import axios from "axios"

// Base URL for your Flask API
const FLASK_API_BASE_URL = "http://127.0.0.1:5000/api/v1/admin/movies/cinemas-screens"

export async function GET() {
  try {
    // TODO: Add authentication check when authentication is implemented
    // For now, this endpoint is open but should be protected in production
    
    // Send a request to the Flask API
    const response = await axios.get(FLASK_API_BASE_URL)

    // Check if the response is successful
    if (response.status === 200) {
      const { data } = response
      return NextResponse.json(data)
    } else {
      return NextResponse.json({ error: "Failed to fetch cinemas and screens" }, { status: response.status })
    }
  } catch (error: any) {
    console.error("Error fetching cinemas and screens:", error)
    if (error.response) {
      return NextResponse.json(
        { error: error.response.data.message || "Failed to fetch cinemas and screens" },
        { status: error.response.status },
      )
    }
    return NextResponse.json({ error: "Internal server error" }, { status: 500 })
  }
}